backend/src/main/java/org/eclipse/tractusx/discoveryfinder/ApiExceptionHandler.java

```diff
@@ -19,22 +19,47 @@
  ********************************************************************************/
 package org.eclipse.tractusx.discoveryfinder;
 
+import java.util.List;
+
+import java.util.Optional;
+import java.util.stream.Collectors;
+
 import org.eclipse.tractusx.discoveryfinder.model.Error;
 import org.eclipse.tractusx.discoveryfinder.model.ErrorResponse;
 import org.eclipse.tractusx.discoveryfinder.service.EntityNotFoundException;
 import org.eclipse.tractusx.discoveryfinder.service.ValidationException;
 import org.springframework.dao.DuplicateKeyException;
+import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
 import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 
 import jakarta.servlet.http.HttpServletRequest;
 
 @ControllerAdvice
 public class ApiExceptionHandler extends ResponseEntityExceptionHandler {
- 
+
+   @Override
+   protected ResponseEntity<Object> handleMethodArgumentNotValid( final MethodArgumentNotValidException ex,
+         final HttpHeaders headers,
+         final HttpStatusCode status, final WebRequest request ) {
+
+      List<Error> errors = ex.getBindingResult()
+            .getFieldErrors()
+            .stream()
+            .map( fieldError -> new Error()
+                  .message( Optional.ofNullable( fieldError.getDefaultMessage() ).orElseGet( () -> "null" ) )
```

Comment:
Please also add the line .putDetailsItem( "parameter",fieldError.getField() ).
Otherwise only the message shown and the enduser does not know which parameter.

Reply:
I added the the parameter with the the affected field. Good hint!